src/plugins/timeConductor/independent/Mode.vue

```diff
@@ -0,0 +1,216 @@
+                .map(this.getClock);
+
+            /*
+         * Populate the modes menu with metadata from the available clocks
+         * "Fixed Mode" is always first, and has no defined clock
+         */
+            this.modes = [undefined]
+                .concat(clocks)
+                .map(this.getModeOptionForClock);
+
+            function isDefinedAndUnique(key, index, array) {
+                return key !== undefined && array.indexOf(key) === index;
+            }
+        },
+
+        getModeOptionForClock(clock) {
+            if (clock === undefined) {
+                const key = 'fixed';
+
+                return {
+                    key,
+                    name: 'Fixed Timespan',
+                    description: 'Query and explore data that falls between two fixed datetimes.',
+                    cssClass: 'icon-tabular',
+                    callBack: () => this.setOption(key)
+                };
+            } else {
+                const key = clock.key;
+
+                return {
+                    key,
+                    name: clock.name,
+                    description: "Monitor streaming data in real-time. The Time "
+              + "Conductor and displays will automatically advance themselves based on this clock. " + clock.description,
+                    cssClass: clock.cssClass || 'icon-clock',
+                    callBack: () => this.setOption(key)
+                };
+            }
+        },
+
+        getClock(key) {
+            return this.openmct.time.getAllClocks().filter(function (clock) {
+                return clock.key === key;
+            })[0];
+        },
+
+        setOption(clockKey) {
+            let key = clockKey;
+            if (clockKey === 'fixed') {
+                key = undefined;
+            }
+
+            let configuration = this.getMatchingConfig({
+                clock: key,
+                timeSystem: this.openmct.time.timeSystem().key
+            });
+
+            if (configuration === undefined) {
+                configuration = this.getMatchingConfig({
+                    clock: key
+                });
+            }
+
+            this.selectedMode = this.getModeOptionForClock(this.getMenuOptions().filter(option => option.clock === clockKey)[0]);
+
+            this.$emit('modeChanged', clockKey);
+        },
+
+        getMatchingConfig(options) {
+            const matchers = {
+                clock(config) {
+                    return options.clock === config.clock;
+                },
+                timeSystem(config) {
+                    return options.timeSystem === config.timeSystem;
+                }
+            };
+
+            function configMatches(config) {
+                return Object.keys(options).reduce((match, option) => {
+                    return match && matchers[option](config);
+                }, true);
+            }
+
+            return this.getMenuOptions().filter(configMatches)[0];
```

Comment:
`.find` instead of `.filter`?

Reply:
No longer using configMatches. Resolving.